core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -100,7 +109,16 @@ public static QuerySourceBuilder parseQuerySource(RestRequest request) {
         if (defaultOperator != null) {
             queryBuilder.defaultOperator(Operator.fromString(defaultOperator));
         }
-        return new QuerySourceBuilder().setQuery(queryBuilder);
+        return queryBuilder;
+    }
+
+    public static SearchSourceBuilder getRestSearchSource(BytesReference sourceBytes, IndicesQueriesRegistry queryRegistry)
+            throws IOException {
+        XContentParser parser = XContentFactory.xContent(sourceBytes).createParser(sourceBytes);
+        QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
```

Comment:
Also, it looks like QueryParseContext.reset() sets parseFieldMatcher to EMPTY again. I think this should be removed so it just sets the existing parseFieldMatcher on the parser?